### ISSUE TITLE
Cannot use domain names with reverse proxy

### ISSUE THREAD

**Issue by Thomas-Philippot** [Yooooomi/your_spotify#246]
Hello, I have succesfully installed the app, but when I try to change my local IP addresses with domain names, i get an error on login
(constantly redirected to login page)

My `docker-compose.yml` file :
```yaml
version: "3"

services:
  server:
    image: yooooomi/your_spotify_server
    restart: always
    ports:
      - "8000:8080"
    links:
      - mongo
    depends_on:
      - mongo
    environment:
      API_ENDPOINT: http://spotify-api.home # This MUST be included as a valid URL in the spotify dashboard (see below)
      CLIENT_ENDPOINT: http://spotify.home
      SPOTIFY_PUBLIC: <private>
      SPOTIFY_SECRET: <private>
  mongo:
    container_name: mongo
    image: mongo:bionic
    volumes:
      - ./your_spotify_db:/data/db

  web:
    image: yooooomi/your_spotify_client
    restart: always
    ports:
      - "3000:3000"
    environment:
      API_ENDPOINT: http://spotify-api.home
```

In nginx proxy manager i added two hosts :

1. `spotify.home` -> wich redirect to my server ip port `3000`
1. `spotify-api.home` -> wich redirect to my server ip port `8000`

Finally on my spotify app the redirect URI is : `http://spotify-api.home/oauth/spotify/callback`

When I access `spotify.home` the frontend loads, but i keep getting redirected to `/login`

The error in console : `GET http://spotify-api.home/version 401 (Unauthorized)`

(I tried adding CORS env var, refreshing browser cache... can't see what is wrong)

**Comment by jack-mil** [comment]
I am also trying a similar setup, let me know if you figure anything out

**Comment by Yooooomi** [comment]
I feel like this is a cookie problem. Having it on a domain without https might be the issue actually.

**Comment by Geda173** [comment]
I've been wrestling with this problem for weeks now and I finally found the solution. Here is my .yml:

`Version: "3"

services:
  server:
    image: yooooomi/your_spotify_server
    restart: always
    ports:
      - "8090:8080"
    links:
      - mongo
    depends_on:
      - mongo
    environment:
      API_ENDPOINT: http://api.subdomain.domain.de
      CLIENT_ENDPOINT: https://subdomain.domain.de
      SPOTIFY_PUBLIC: secret key
      SPOTIFY_SECRET: secret key
  mongo:
    container_name: mongo
    image: mongo:6
    volumes:
      - ./your_spotify_db:/data/db

  web:
    image: yooooomi/your_spotify_client
    restart: always
    ports:
      - "3090:3000"
    environment:
      API_ENDPOINT: http://api.subdomain.domain.de'


Note the slight difference in API_ENDPOINT and CLIENT_ENDPOINT: API uses http, CLIENT https. This has finally solved my problem as the login always came back with an error message. Maybe this helps.

**Comment by Dav-Git** [comment]
Hi, I'm having a similar problem however @Geda173 's approach did not solve the problem for me.
From the error it looks like somehow a CORS error is to blame: 
![image](https://github.com/Yooooomi/your_spotify/assets/57032623/00c6b74b-70b5-48c7-86e6-7b1fdefd7ec2)

The server logs seem to suggest that it knows my spotify account as it has some interaction with the DB talking about it.
![image](https://github.com/Yooooomi/your_spotify/assets/57032623/1d0c6340-16ee-4214-83ef-cdb5ae9be27c)


I have tried setting the ``CORS`` env var to:
- my ``CLIENT_ENDPOINT`` URL 
- "all"
- "``CLIENT_ENDPOINT``, ``API_ENDPOINT``"
- nothing

with no success.

I do however remember CORS being a sneaky little thing from the early days of my (I have bailed on that a long time ago) web dev career.
